product_pricelist_margins/models/product_pricelist_item.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    cost = fields.Float(
+        related="product_tmpl_id.standard_price",
+        digits="Product Price",
+    )
+    margin = fields.Float(
+        compute="_compute_margin",
+        digits="Product Price",
+    )
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        compute="_compute_margin",
+    )
+
+    @api.depends(
+        "compute_price",
+        "applied_on",
+        "percent_price",
+        "base",
+        "price_discount",
+        "price_surcharge",
+        "price_round",
+        "price_min_margin",
+        "product_tmpl_id",
+        "cost",
+        "min_quantity",
+    )
+    def _compute_margin(self):
+        for item in self.filtered(
```

Comment:
Suggestion: You could avoid to loop twice in self.
Simply do something like : 
for item in self:
    if item in ("1_product", "0_product_variant"):
        Do something
    else:
        item.margin = 0
        item.margin_percent = 0

Reply:
Hi @santostelmo , I have updated it